### PR TITLE
refactored datanode and journalnode ports into attributes.

### PR DESCRIPTION
**Pull request by rashidaligee**
This item was somehow landed in Ambari Epic, instead of moving it around, just implemented it.

### CODE REVIEW

File: cookbooks/bcpc-hadoop/attributes/hdfs.rb [hunk 1]
```diff
@@ -13,10 +13,14 @@
 # # set to nil to calculate dynamically based on available memory
 default['bcpc']['hadoop']['namenode']['xmn']['max_size'] = 128
 default['bcpc']['hadoop']['namenode']['xmx']['max_ratio'] = 0.25
-default['bcpc']['hadoop']['namenode']['jmx']['port'] = 10111
+default['bcpc']['hadoop']['namenode']['jmx']['port'] = 10_111
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/attributes/hdfs.rb]: I don't think we should be using JMX port anymore? (We use jmxtrans_agent to send data so we don't need to network poll or expose JMX.)

File: cookbooks/bcpc-hadoop/recipes/hdfs_site.rb [hunk 2]
```diff
@@ -211,13 +211,13 @@
       jn_properties =
         {
          'dfs.journalnode.rpc-address' =>
-           node[:bcpc][:floating][:ip] + ':8485',
+           node[:bcpc][:floating][:ip] + ':' + node['bcpc']['hadoop']['journalnode']['rpc']['port'].to_s,
```
Comment (vt0r) [inline cookbooks/bcpc-hadoop/recipes/hdfs_site.rb]: For this and other concatenated strings below, could we use interpolation instead? It's arguably *slightly* easier to read, the `.to_s` happens implicitly, and it's also slightly faster in benchmarks (interpolation is faster already, plus it now becomes a single string).

Example:
``` ruby
"#{node['bcpc']['floating']['ip']}:#{node['bcpc']['hadoop']['journalnode']['rpc']['port']}",
```

File: cookbooks/bcpc-hadoop/attributes/hdfs.rb (hunk 1 above)
Comment (rashidaligee) [reply cookbooks/bcpc-hadoop/attributes/hdfs.rb]: I tried to please rubocop here. My change is just to add `_` in a number.

**Comment by rashidaligee** [comment]
- Interpolation is used to concatenate attributes.

File: cookbooks/bcpc-hadoop/recipes/hdfs_site.rb [hunk 3]
```diff
@@ -262,8 +262,8 @@
 
          'dfs.namenode.shared.edits.dir' =>
            'qjournal://' +
-           zk_hosts.map{ |s| float_host(s[:fqdn]) + ":8485" }.join(";") +
-           '/' + node.chef_environment,
+           zk_hosts.map { |s| float_host(s[:fqdn]) + ':' + node['bcpc']['hadoop']['journalnode']['rpc']['port'].to_s }
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/recipes/hdfs_site.rb]: Can this be an interpolation too?

**Pull request merged**